blockchain/txpool.go

```diff
@@ -159,19 +166,36 @@ func (pool *TxPool) verifyDoubleSpend(txn *Transaction) error {
 
 //check and add to sidechain tx pool
 func (pool *TxPool) verifyDuplicateSidechainTx(txn *Transaction) error {
-	if txn.IsWithdrawTx() {
-		witPayload := txn.Payload.(*PayloadWithdrawAsset)
-		for _, hash := range witPayload.SideChainTransactionHash {
-			_, ok := pool.sidechainTxList[hash]
-			if ok {
-				return errors.New("duplicate sidechain tx detected")
-			}
+	witPayload := txn.Payload.(*PayloadWithdrawAsset)
+	for _, hash := range witPayload.SideChainTransactionHash {
```

Comment:
use witPayload, ok := txn.Payload.(*PayloadWithdrawAsset) to prevent nil pointer panic